app/controllers/api/connections_controller.rb

```diff
@@ -92,6 +92,18 @@ def update_match
                         return render(json: error, status: 500)
                     end
 
+                    # we also need to mark the own user's notification as read since they responded
+                    own_notification = Notification.find_by(match_id: @request.id, person_id: @current_user.id, notification_type: "received_request")
+                    
+                    if own_notification
+                        unless own_notification.update(read: true)
+                            return render_json_500(own_notification.errors.full_messages)
+                        end
+                    else
+                        # Shouldn't happen, notification not found
+                        puts "Error: there was no notification"
```

Comment:
we should have no "puts" anywhere. It usually means that something will fail silently. So

* to manage errors that should break the process: I'd recommend raising an exception and capture it at an higher level like a controller. Then report it using sentry for example and answer the client with a proper message.
* to simply log things, use rails logger so we can control the level of verbosity

Reply:
That situation is a case that is not the user's fault (hence not raising an exception or returning a status 500 error) and it's not a very serious one either, so the point was indeed to fail silently in a way (only logging out a note), but the logger seems like a better option, I just didn't know about it!